### PR TITLE
SONARFLEX-53, SONARFLEX-54, SONARFLEX-50 Upgrade sonar-plugin-api from 3.7 to 4.5.2 

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java [hunk 1]
```diff
@@ -69,17 +68,23 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
...
     while (pack.getNext() != null) {
       Map<String, CoverageMeasuresBuilder> builderByFilename = Maps.newHashMap();
       collectFileMeasures(pack.descendantElementCursor("class"), builderByFilename);
+      FilePredicates predicates = fileSystem.predicates();
 
       for (Map.Entry<String, CoverageMeasuresBuilder> entry : builderByFilename.entrySet()) {
-        Resource resource = getResourceForFileRelativeName(entry.getKey(), fileSystem, project);
 
-        if (resource != null) {
+        String fileName = entry.getKey().startsWith(File.separator) ? entry.getKey() : File.separator + entry.getKey();
```
Comment (SonarTech) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java [hunk 2]
```diff
@@ -69,17 +68,23 @@ public void stream(SMHierarchicCursor rootCursor) throws XMLStreamException {
...
     while (pack.getNext() != null) {
       Map<String, CoverageMeasuresBuilder> builderByFilename = Maps.newHashMap();
       collectFileMeasures(pack.descendantElementCursor("class"), builderByFilename);
+      FilePredicates predicates = fileSystem.predicates();
 
       for (Map.Entry<String, CoverageMeasuresBuilder> entry : builderByFilename.entrySet()) {
-        Resource resource = getResourceForFileRelativeName(entry.getKey(), fileSystem, project);
 
-        if (resource != null) {
+        String fileName = (entry.getKey().startsWith(File.separator) ? entry.getKey() : File.separator + entry.getKey());
```
Comment (SonarTech) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
MAJOR Add parentheses to make the operator precedence explicit. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by vilchik-elena** [comment]
LGTM!

**Comment by lindamartin** [comment]
@vilchik-elena indeed! Updated; thanks.

File: sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java [hunk 3]
```diff
@@ -20,29 +20,28 @@
...
-import javax.annotation.Nullable;
 import javax.xml.stream.XMLStreamException;
 import java.io.File;
 import java.text.ParseException;
 import java.util.Map;
 
-import static java.util.Locale.ENGLISH;
-import static org.sonar.api.utils.ParsingUtils.parseNumber;
-
 public class CoberturaReportPasrer {
```
Comment (vilchik-elena) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java]: @lindamartin misprint in class name CoberturaReportPa**sr**er

Comment (lindamartin) [inline sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/cobertura/CoberturaReportPasrer.java]: @vilchik-elena indeed thanks, fixed in 9b937d42451e8db944cd344c1ffa046fb9357086.

**Pull request merged**